lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb

```diff
@@ -225,7 +245,9 @@ def correcting_scan(scan_run_options, batch, reportnamer)
           if try_count < @try_count
             @retry_total_count += 1
             scan_options.delete(:code_coverage)
-            scan_options[:only_testing] = info[:failed].map(&:shellsafe_testidentifier)
+            last_failed_tests = info[:failed].map(&:shellsafe_testidentifier)
+            last_failed_tests = last_failed_tests.map(&:strip_testcase) if @invocation_based_tests
```

Comment:
One last thing that I realized, this will most likely create an array of several repeated identifiers. i.e. `[ 'testable/testsuite1/testcaseA', 'testable/testsuite1/testcaseB'] -> [ 'testable/testsuite1', 'testable/testsuite1' ]`. We should use `uniq` to get `[ 'testable/testsuite1/testcaseA', 'testable/testsuite1/testcaseB'].uniq -> [ 'testable/testsuite1' ]`
```suggestion
            last_failed_tests = last_failed_tests.map(&:strip_testcase).uniq if @invocation_based_tests
```

Reply:
I used uniq on the following line https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/pull/102/files#diff-7ab9d5a553f6236cc7846db28ad0060fR250

Do you think I should use it only for invocation based tests?

Reply:
No, I think that is fine. I forgot about that. It is unlikely that XCTests will have duplicates, but it doesn't hurt.